### PR TITLE
Add Safari Doctors Youth Health Ambassadors app flavor.

### PR DESCRIPTION
**Pull request by benkags**
contributes medic/safari-doctors#20

### CODE REVIEW

**Comment by antonykhaemba** [comment]
Hi @eljhkrr, I tagging you here since the apk is needed by tomorrow - sorry I am not able to assign this PR to you.

**Comment by eljhkrr** [comment]
@benkags Confirm whether signing secrets have been setup within the repository

**Comment by eljhkrr** [comment]
No worries @antonykhaemba, I've assigned myself

**Comment by benkags** [comment]
@eljhkrr The secrets have not been set up. Let me now how I can share them securely

{1C06F493-6333-44F9-9E25-95A9E9B4560E}

**Comment by eljhkrr** [comment]
Reassigned to Wambua as I'll be away today

**Comment by billwambua** [comment]
@benkags can you [upload to 1password](https://support.1password.com/files/) and invite @eljhkrr and me.

**Comment by benkags** [comment]
@billwambua I have shared the credentials via 1password. I have requested @eljhkrr 's email via email so that I can add him as well.

**Comment by billwambua** [comment]
@benkags I tried opening the link you shared on email but it says "Not available"

Screenshot from 2024-09-16 13-25-40

**Comment by billwambua** [comment]
@benkags for the 1Password invitation, you invited the wrong email. You are using the new shortened domain instead of the old medic domain which is what I created my 1Password account with.

This means I have to create a new account to join your invitation.

I have communicated with @antonykhaemba on the exact details you can use if you still don't understand what you need to do from the above. 

He will forward to you.

**Comment by benkags** [comment]
Hi Bill, the BitWarden share expired.

@billwambua, I've shared with wa*@medicmobile.org address. Let me know if you can access now.

**Comment by billwambua** [comment]
@benkags I'm in the workspace you have invited me to, but I can't see any Safari Doctors files.

**Comment by benkags** [comment]
It was pending approval.

'View vault' so see the .env file uploaded.

safari doctors 1password vault

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -575,6 +575,31 @@ jobs:
...
+      uses: maierj/fastlane-action@v1.4.0
+      with:
+        lane: build
+        options: '{ "flavor": "safaridoctors_ke_yha" }'
+      env:
+        ANDROID_KEYSTORE_PATH: safaridoctors_ke_yha.keystore
+        ANDROID_KEYSTORE_PASSWORD: ${{ secrets.ANDROID_KEYSTORE_PASSWORD_SAFARIDOCTORS_KE_YHA }}
+        ANDROID_KEY_PASSWORD: ${{ secrets.ANDROID_KEY_PASSWORD_SAFARIDOCTORS_KE_YHA }}
+
+    - name: Bundle new_brand
```
Comment (billwambua) [inline .github/workflows/publish.yml:594]: Rename all instances of `new_brand` to `safaridoctors_kenya_yha` for consistency.

Make the naming consistent. You have `SAFARIDOCTORS_KE_YHA` in some places and `safaridoctors_kenya_yha` in others. Choose between `ke` and `kenya`.

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -575,6 +575,31 @@ jobs:
         ANDROID_KEYSTORE_PASSWORD: ${{ secrets.ANDROID_KEYSTORE_PASSWORD_GANDAKI_NE }}
         ANDROID_KEY_PASSWORD: ${{ secrets.ANDROID_KEY_PASSWORD_GANDAKI_NE }}
 
+    - name: Unpack secrets safaridoctors_ke_yha
+      env:
+        ANDROID_SECRETS_KEY: ${{ secrets.ANDROID_SECRETS_KEY_SAFARIDOCTORS_KE_YHA }}
+        ANDROID_SECRETS_IV: ${{ secrets.ANDROID_SECRETS_IV_SAFARIDOCTORS_KE_YHA }}
+      run: make org=new_brand keydec
+
+    - name: Assemble new_brand
```
Comment (billwambua) [inline .github/workflows/publish.yml:584]: Rename `new_brand` to `safaridoctors_kenya_yha` for consistency.

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -575,6 +575,31 @@ jobs:
         ANDROID_KEYSTORE_PASSWORD: ${{ secrets.ANDROID_KEYSTORE_PASSWORD_GANDAKI_NE }}
         ANDROID_KEY_PASSWORD: ${{ secrets.ANDROID_KEY_PASSWORD_GANDAKI_NE }}
 
+    - name: Unpack secrets safaridoctors_ke_yha
+      env:
+        ANDROID_SECRETS_KEY: ${{ secrets.ANDROID_SECRETS_KEY_SAFARIDOCTORS_KE_YHA }}
+        ANDROID_SECRETS_IV: ${{ secrets.ANDROID_SECRETS_IV_SAFARIDOCTORS_KE_YHA }}
+      run: make org=new_brand keydec
```
Comment (billwambua) [inline .github/workflows/publish.yml:582]: Rename `new_brand` to `safaridoctors_kenya_yha` for consistency.

**Review by billwambua — CHANGES_REQUESTED** [review]

**Comment by billwambua** [comment]
@benkags I don't have the appropriate permissions to add the secrets to Github Actions. 

@alexosugo can sort that.

**Comment by benkags** [comment]
@billwambua, thanks. I've standardized the naming.

**Review by billwambua — APPROVED** [review]
LGTM.

Go ahead with setting the CI variables.

**Review by jkuester — APPROVED** [review]
LGTM!

**Comment by benkags** [comment]
@jkuester I have shared the credentials in 1password to jk*@medic.org.

**Comment by jkuester** [comment]
@benkags I received an email invite to make an account for your 1Password.  I did that (and my user is `jkuester`.)  But I cannot see any existing vault items.  Could you try sharing them with my `jkuester` account?

**Comment by benkags** [comment]
Access was pending approval. Pleas try now. Check the vault items ford the .env file.

**Comment by mrjones-plip** [comment]
I've added these secrets:
* `ANDROID_KEYSTORE_PASSWORD_SAFARIDOCTORS_KENYA_YHA`
* `ANDROID_KEY_PASSWORD_SAFARIDOCTORS_KENYA_YHA`
* `ANDROID_SECRETS_IV_SAFARIDOCTORS_KENYA_YHA`
* `ANDROID_SECRETS_KEY_SAFARIDOCTORS_KENYA_YHA`

image

**Comment by jkuester** [comment]
Okay, I merged this to `master` and pushed a tag.  The release artifacts (`.apk` and `aab`) are available [here](https://github.com/medic/cht-android/releases/tag/untagged-b1d7f6de78d292712e37)  (look for the files `cht-android-v1.4.0-safaridoctors_kenya_yha-publish.1-safaridoctors_kenya_yha-*`).

**Comment by benkags** [comment]
I don't have access to [this link](https://github.com/medic/cht-android/releases/tag/untagged-b1d7f6de78d292712e37). Are there additional privileges required. Is there another way to access the APKs before a [release tag](https://github.com/medic/cht-android/releases) is created?

**Pull request merged**